File: src/server/Application/Admissions/ViewModels/ApplicantDashboardDto.cs
using System.Text.Json.Serialization;

namespace ERP.Application.Admissions.ViewModels;

public sealed record ApplicantDashboardDto(
    ApplicantProfileDto Profile,
    IReadOnlyCollection<ApplicantDocumentStatusDto> Documents,
    IReadOnlyCollection<ApplicantNotificationDto> Notifications,
    ApplicantDashboardApplicationDto Application,
    ApplicantDashboardPaymentDto Payment,
    ApplicantCourseSelectionSummaryDto? CourseSelection = null,
    ApplicantDashboardOfferDto? Offer = null);

/// <summary>Elective line for dashboard: code, display name, and full catalog text (tooltip).</summary>
public sealed record ApplicantElectiveSubjectDto(
    [property: JsonPropertyName("code")] string Code,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("description")] string? Description);

/// <summary>Populated when application is submitted and application fee payment is completed.</summary>
public sealed record ApplicantCourseSelectionSummaryDto(
    string PreferredShiftCode,
    string PreferredShiftLabel,
    string MajorSubject,
    string MinorSubject,
    ApplicantElectiveSubjectDto Mdc,
    ApplicantElectiveSubjectDto Aec,
    ApplicantElectiveSubjectDto Sec,
    ApplicantElectiveSubjectDto Vac,
    DateTime? ApplicationFeePaidOnUtc,
    DateTime? DraftLastUpdatedOnUtc);

public sealed record ApplicantProfileDto(
    Guid AccountId,
    string UniqueId,
    string FullName,
    DateOnly DateOfBirth,
    string Gender,
    string Email,
    string MobileNumber,
    string Shift,
    string? PhotoUrl,
    DateTime CreatedOnUtc);

public sealed record ApplicantDocumentStatusDto(
    string Name,
    string Status,
    string Description,
    bool IsComplete);

public sealed record ApplicantNotificationDto(
    string Title,
    string Message,
    DateTime CreatedOnUtc);

public sealed record ApplicantDashboardApplicationDto(
    bool IsSubmitted,
    bool CoursesLocked,
    string Status,
    IReadOnlyCollection<ApplicantDashboardApplicationStepDto> Steps);

public sealed record ApplicantDashboardApplicationStepDto(
    string Key,
    string Title,
    bool IsComplete,
    string? Description);

public sealed record ApplicantDashboardPaymentDto(
    decimal AmountDue,
    decimal AmountPaid,
    string Status,
    bool CanPay,
    string? TransactionId = null);

public sealed record ApplicantDashboardOfferDto(
    Guid OfferId,
    string ApplicationNumber,
    int MeritRank,
    string Shift,
    string MajorSubject,
    string Status,
    DateTime OfferDate,
    DateTime ExpiryDate,
    DateTime? AcceptedOnUtc,
    DateTime? RejectedOnUtc,
    bool IsExpired,
    int DaysUntilExpiry);

File: src/client/shared/data/src/lib/dtos/applicant-dashboard.dto.ts
export interface ApplicantProfileDto {
  accountId: string;
  uniqueId: string;
  fullName: string;
  dateOfBirth: string;
  gender: string;
  email: string;
  mobileNumber: string;
  shift: string;
  photoUrl?: string | null;
  createdOnUtc: string;
}

export interface ApplicantDocumentStatusDto {
  name: string;
  status: string;
  description: string;
  isComplete: boolean;
}

export interface ApplicantNotificationDto {
  title: string;
  message: string;
  createdOnUtc: string;
}

/** Elective row: subject name + code; description = full catalog line for tooltips. */
export interface ApplicantElectiveSubjectDto {
  code: string;
  name: string;
  description: string | null;
}

/** Present when application is submitted and application fee payment completed. */
export interface ApplicantCourseSelectionSummaryDto {
  preferredShiftCode: string;
  preferredShiftLabel: string;
  majorSubject: string;
  minorSubject: string;
  mdc: ApplicantElectiveSubjectDto;
  aec: ApplicantElectiveSubjectDto;
  sec: ApplicantElectiveSubjectDto;
  vac: ApplicantElectiveSubjectDto;
  applicationFeePaidOnUtc: string | null;
  draftLastUpdatedOnUtc: string | null;
}

export interface ApplicantDashboardDto {
  profile: ApplicantProfileDto;
  documents: ApplicantDocumentStatusDto[];
  notifications: ApplicantNotificationDto[];
  application: ApplicantDashboardApplicationDto;
  payment: ApplicantDashboardPaymentDto;
  /** Present when application is submitted and fee payment is completed. */
  courseSelection?: ApplicantCourseSelectionSummaryDto | null;
}

export interface ApplicantDashboardApplicationDto {
  isSubmitted: boolean;
  coursesLocked: boolean;
  status: string;
  steps: ApplicantDashboardApplicationStepDto[];
}

export interface ApplicantDashboardApplicationStepDto {
  key: string;
  title: string;
  isComplete: boolean;
  description?: string;
}

export interface ApplicantDashboardPaymentDto {
  transactionId?: string | null;
  amountDue: number;
  amountPaid: number;
  status: string;
  canPay: boolean;
}
